src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -2258,7 +2259,13 @@ private static Task WriteJsonResponse(HttpResponse response, object? value, Json
         // https://github.com/dotnet/aspnetcore/issues/43894
         // https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-polymorphism
         return HttpResponseJsonExtensions.WriteAsJsonAsync(response, value, value is null ? typeof(object) : value.GetType(), options, default);
```

Comment:
```suggestion
        Debug.Assert(value is null || !value.GetType().IsValueType, "WriteJsonResponse boxes value type values. WriteJsonResponseOfT should be used to avoid boxing.");
        return HttpResponseJsonExtensions.WriteAsJsonAsync(response, value, value is null ? typeof(object) : value.GetType(), options, default);
```